### ISSUE TITLE
Motion detector not working

### ISSUE THREAD

**Issue by dexdeadly** [elad-bar/ha-shinobi#18]
I have motion detector enabled for a camera, I have the region setup and when I am watching the logs via the triangle in shinobie I do see the following. 

`
hP4K31upsS : Ot1xEiBbDf
Event Occurred 2 minutes ago
plug : built-in
name : multipleRegions
reason : motion
confidence : 6
matrices :
0 :
confidence : 5
width : 96
height : 196
x : 472
y : 215
tag : LAI7T
imgHeight : 480
imgWidth : 640
`

So to me this looks like the motion detection is working on the Shinobi side.  The HA side, I see the motion detector and its set as an entity to the device.  However it only ever says clear.  Nothing seems to change that.  I'd like to get this working and also get it working with deepstack-face but I can't seem to get just basic motion detector working.

**Comment by elad-bar** [comment]
Not sire I understand the issue,
You have configured the motion detection and object detection of face, 
You are not getting the event of motion, right?

**Comment by dexdeadly** [comment]
Yes, when I go to my camera and setup under advanced, right now I have just PRIMARY ENGINE : PIXEL ARRAY because I have the use built in set to yes.  The camera in Shinobi shows the red dotted box around me as I move into the camera frame and I see that message in the logs.  But HA never sees a trigger for motion.  It just always stays clear.

**Comment by elad-bar** [comment]
can you please change the log level of the component to debug and post the logs once you have that kind of issue?
it will assist understanding the cause

thanks

**Comment by elad-bar** [comment]
@dexdeadly , any update?

thanks

**Comment by dexdeadly** [comment]
My apologies, I just recently switched over to home assistant os and I'm still not getting it.  Let me see if I can switch the home assistant component level and check.

**Comment by dexdeadly** [comment]
Hello.  So just to confirm.  I do not have the DeepStack-Object or DeepStack-Face setup.  I would like to set those up.  However I was just currently testing with just the motion detector and was hoping to see the event trigger.  Do I need to have DeepStack-Face or Object setup in order for that sensor to show anything within Home Assistant?  Is there instructions or a video to watch for the full setup of using this plugin?

**Comment by elad-bar** [comment]
No, you just need the shinobi video to be configured working with motion detection.

Face or object plugins (no matter which plugin of shinobi) will publish detection as an event.

To sum it up, once you have set the motion detection within shinobi, it should seamlessly work for you,
If that is not the case, i will try to assist you

**Comment by dexdeadly** [comment]
Ok, so I have the logging set to debug using the following in my configuration.yaml

`# Logging Setup
logger:
  default: info
  logs:
    custom_components.shinobi: debug`

When the motion detection is triggered in shinobi I can see the monitor log display like I showed in the first message.  However there is nothing that I can see in the log showing any sort of attempt to send to HA.  Now if I do a restart of HA and the component loads I do see this in the log.

`2021-11-06 13:31:30 INFO (MainThread) [custom_components.shinobi.binary_sensor] Added new Shinobi Video Living/Dining Room Sound
2021-11-06 13:31:30 INFO (MainThread) [custom_components.shinobi.binary_sensor] Added new Shinobi Video Living/Dining Room Motion
2021-11-06 13:31:30 INFO (MainThread) [custom_components.shinobi.binary_sensor] Added new Shinobi Video Girls Room Sound
2021-11-06 13:31:30 INFO (MainThread) [custom_components.shinobi.camera] Added new Shinobi Video Living/Dining Room
2021-11-06 13:31:30 INFO (MainThread) [custom_components.shinobi.camera] Added new Shinobi Video Girls Room
2021-11-06 13:31:30 INFO (MainThread) [custom_components.shinobi.api.shinobi_websocket] Starting to listen connected
2021-11-06 13:31:30 DEBUG (MainThread) [custom_components.shinobi.api.shinobi_websocket] No message handler available, Message: 0{"sid":"1Q3NnTGJ1fJ-JrpnAALn","upgrades":[],"pingInterval":25000,"pingTimeout":5000}
2021-11-06 13:31:30 DEBUG (MainThread) [custom_components.shinobi.api.shinobi_websocket] WebSocket connection state changed to ready
2021-11-06 13:31:30 DEBUG (MainThread) [custom_components.shinobi.api.shinobi_websocket] Sending message, Data: 42["f", {"auth": "REDACTED", "f": "init", "ke": "REDACTED", "uid": "REDACTED"}], Connected: True
`

Then this is repeated multiple times but just looks like health checks.

`2021-11-06 13:39:25 DEBUG (MainThread) [custom_components.shinobi.managers.home_assistant] Updating @2021-11-06 17:39:25.538007+00:00
2021-11-06 13:39:25 INFO (MainThread) [custom_components.shinobi.api.shinobi_api] Updating data from Shinobi Video Server (Shinobi Video)
2021-11-06 13:39:25 DEBUG (MainThread) [custom_components.shinobi.api.shinobi_api] Retrieving camera list
2021-11-06 13:39:25 DEBUG (MainThread) [custom_components.shinobi.api.shinobi_api] GET http://10.1.1.25:8080/P7mMVlTCB0iXsKRn9FVI3iAnbjOc9G/monitor/hP4K31upsS
2021-11-06 13:39:25 DEBUG (MainThread) [custom_components.shinobi.api.shinobi_api] Status of http://10.1.1.25:8080/P7mMVlTCB0iXsKRn9FVI3iAnbjOc9G/monitor/hP4K31upsS: 200
2021-11-06 13:39:45 DEBUG (MainThread) [custom_components.shinobi.api.shinobi_websocket] Pinging
2021-11-06 13:39:45 DEBUG (MainThread) [custom_components.shinobi.api.shinobi_websocket] Sending message, Data: 2, Connected: True
2021-11-06 13:39:45 DEBUG (MainThread) [custom_components.shinobi.api.shinobi_websocket] Pong message received
2021-11-06 13:40:10 DEBUG (MainThread) [custom_components.shinobi.api.shinobi_websocket] Pinging
2021-11-06 13:40:10 DEBUG (MainThread) [custom_components.shinobi.api.shinobi_websocket] Sending message, Data: 2, Connected: True
2021-11-06 13:40:10 DEBUG (MainThread) [custom_components.shinobi.api.shinobi_websocket] Pong message received
2021-11-06 13:40:25 DEBUG (MainThread) [custom_components.shinobi.managers.home_assistant] Updating @2021-11-06 17:40:25.538324+00:00
2021-11-06 13:40:25 INFO (MainThread) [custom_components.shinobi.api.shinobi_api] Updating data from Shinobi Video Server (Shinobi Video)
2021-11-06 13:40:25 DEBUG (MainThread) [custom_components.shinobi.api.shinobi_api] Retrieving camera list
2021-11-06 13:40:25 DEBUG (MainThread) [custom_components.shinobi.api.shinobi_api] GET http://10.1.1.25:8080/P7mMVlTCB0iXsKRn9FVI3iAnbjOc9G/monitor/hP4K31upsS
2021-11-06 13:40:25 DEBUG (MainThread) [custom_components.shinobi.api.shinobi_api] Status of http://10.1.1.25:8080/P7mMVlTCB0iXsKRn9FVI3iAnbjOc9G/monitor/hP4K31upsS: 200
2021-11-06 13:40:35 DEBUG (MainThread) [custom_components.shinobi.api.shinobi_websocket] Pinging
2021-11-06 13:40:35 DEBUG (MainThread) [custom_components.shinobi.api.shinobi_websocket] Sending message, Data: 2, Connected: True
2021-11-06 13:40:35 DEBUG (MainThread) [custom_components.shinobi.api.shinobi_websocket] Pong message received
2021-11-06 13:41:00 DEBUG (MainThread) [custom_components.shinobi.api.shinobi_websocket] Pinging
2021-11-06 13:41:00 DEBUG (MainThread) [custom_components.shinobi.api.shinobi_websocket] Sending message, Data: 2, Connected: True
2021-11-06 13:41:00 DEBUG (MainThread) [custom_components.shinobi.api.shinobi_websocket] Pong message received`

Let me know what else to try and I can get it for you.

**Comment by elad-bar** [comment]
Hi, thanks for the logs.
As I understand,  you have 2 cameras, 
Living/Dining room with sound and motion detection, 
Girl's room with sound detection only,
Right?

**Comment by dexdeadly** [comment]
Correct, I didn't setup the motion on the girls room camera just yet.  I wanted to ensure the correct setup on one camera before I repeat the process.

**Comment by elad-bar** [comment]
Sorry for the late response,  there should be log messages with description of the event in HA log, but I cannot see them,
Will try to release a newer version with more logs later

**Comment by elad-bar** [comment]
No need for additional version as this one covering all scenarios. 

Can you please reactivate the log level in debug, monitor both log of ha and shinobi, trigger a motion alert, make sure it was logged in both system and post the logs?

Thanks

**Comment by elad-bar** [comment]
Any updates?

**Comment by jarrodsfarrell** [comment]
Don't want this thread to die since I'm having similar issues with motion not being forwarded to HA.

Overarching setup for our house is all the still cameras use the built in Pixel Array motion detection that's then followed by object detection with a defined region to avoid motion by trees or the OSD "burned" into the video.

So added the following to my `configuration.yaml`
```yaml
# Diagnosing Shinobi
logger:
  default: warning
  logs:
    custom_components.shinobi: debug
```
Restart HA and I see some logs from the integration. No glaring errors: just it recognizing, connecting, and enumerating the cameras along with occasional refresh. So I pick a camera to walk in front: the front porch. And...nothing. No logs generated by the integration while watching the camera off my phone I can see it drawing motion and a bounding box for myself.

Investigating I'm noticing the stream key the integration is using. This case `orG3...` which points to a restrictive API that only allows viewing the streams and snapshots from a previous test project. It didn't even try the `ojqS...` I made for it which has almost everything but deleting or controlling the monitors.

For testing I'll clean out the API keys and make a singular, all access API key on the account (but accidentally leave out delete video,) refresh the integration and now I see it talking over the web-socket connection using the API key, but disconnecting occasionally because it's not replying back to a ping I guess.
```
No payload handler available, Payload: ['ping', {'beat': 1}]
```
I do restart HA again since I notice I can't view the cameras anymore which brings them back. I also notice the websocket connection is working correctly this time and I see messages of it sending a ping itself on it's own timer which seems to be keeping the connection alive.

Now attempt two.
```
[....shinobi.api.shinobi_websocket] Payload received, Data: {'f': 'detector_trigger', 'id': 'XXX', 'ke': 'YYY', 'details': {'plug': 'built-in', 'name': 'Movement', 'reason': 'motion', 'confidence': 3, 'matrices': [{'x': 0, 'y': 98, 'width': 639, 'height': 381, 'tag': 'Movement'}], 'imgHeight': '480', 'imgWidth': '640'}, 'doObjectDetection': True}
[....shinobi.managers.event_manager] Set motion state: {'name': 'Movement', 'plug': 'built-in', 'reason': 'motion', 'state': 'on', 'timestamp': 1646026859.583889, 'topic': 'YYY/XXX'} for YYY/XXX
```
Hey, something!
![image](https://user-images.githubusercontent.com/44697494/155930597-eb3d0ee2-6be4-4942-8c3e-3d9c60ca3af9.png)

So my conclusion is that either prompt for a username and password to select a API key to use or provide a API key yourself when adding/configuring the integration. Plus some code polishing since there's a few quirks in re-configuring the integration.

----
**TL;DR:** This integration picks the last API key even if the key has no or insufficient access. It should prompt to select a API to use or let you flatly give one for it to use.

**Comment by elad-bar** [comment]
I think I fixed the same issue while adding the support for dashboard-v3 that requires the same functionality of ping/pong,
please check if it was solved with v1.1.22 and let me know.

thanks

**Comment by elad-bar** [comment]
@dexdeadly  @jarrodsfarrell  - can you pls confirm the issue still take place or resolved?

thanks

**Comment by elad-bar** [comment]
closing the issue, if being reproduced, feel free to reopen
